### PR TITLE
Run shellcheck

### PR DESCRIPTION
**Pull request by axic**
Uses https://github.com/koalaman/shellcheck.

Should consider just installing shellcheck and not using the orb.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 version: 2.1
 
+orbs:
+  shellcheck: circleci/shellcheck@1.3.16
```
Comment (axic) [inline circle.yml:4]: This is actually just a pip package, we should include it in `cpp-build-env` and make it part of the lint step.

**Review by chfast — APPROVED** [review]

File: circle.yml (hunk 1 above)
Comment (chfast) [reply circle.yml]: Not a pip package, but added to cpp-build-env:15-lint and cpp-build-env:15-clang-10.

File: circle.yml [hunk 2]
```diff
@@ -215,6 +215,10 @@ jobs:
     executor: linux-clang-latest
     steps:
       - checkout
+      - run:
+          name: "Run shellcheck"
+          command: |
+            git ls-files '*.sh' | xargs shellcheck --external-sources
```
Comment (chfast) [inline circle.yml:221]: Will this fail on error?

Comment (axic) [reply circle.yml:221]: Yes, I checked it with removing one quote from `apply_patches.sh`.

**Review by chfast — APPROVED** [review]

**Pull request merged**